controllers/activegate/capability/reconciler.go

```diff
@@ -20,39 +20,48 @@ import (
 )
 
 const (
-	Module            = "routing"
-	StatefulSetSuffix = "-" + Module
-	capabilityName    = "MSGrouter"
-	containerPort     = 9999
-	DTDNSEntryPoint   = "DT_DNS_ENTRY_POINT"
+	containerPort   = 9999
+	DTDNSEntryPoint = "DT_DNS_ENTRY_POINT"
 )
 
 type Reconciler struct {
-	*capability.Reconciler
+	*activegate.Reconciler
 	log logr.Logger
+	*Capability
 }
 
-func NewReconciler(clt client.Client, apiReader client.Reader, scheme *runtime.Scheme, dtc dtclient.Client, log logr.Logger,
+func NewReconciler(capability *Capability, clt client.Client, apiReader client.Reader, scheme *runtime.Scheme, dtc dtclient.Client, log logr.Logger,
 	instance *dynatracev1alpha1.DynaKube, imageVersionProvider dtversion.ImageVersionProvider) *Reconciler {
-	baseReconciler := capability.NewReconciler(
+	baseReconciler := activegate.NewReconciler(
 		clt, apiReader, scheme, dtc, log, instance, imageVersionProvider,
-		&instance.Spec.RoutingSpec.CapabilityProperties, Module, capabilityName, "")
-	baseReconciler.AddOnAfterStatefulSetCreateListener(addDNSEntryPoint(instance))
-	baseReconciler.AddOnAfterStatefulSetCreateListener(setCommunicationsPort(instance))
-	baseReconciler.AddOnAfterStatefulSetCreateListener(setReadinessProbePort(instance))
+		capability.Properties, capability.ModuleName, capability.CapabilityName, capability.ServiceAccountOwner)
+
+	if capability.Configuration.SetDnsEntryPoint {
+		baseReconciler.AddOnAfterStatefulSetCreateListener(addDNSEntryPoint(instance, capability.ModuleName))
+	}
+
+	if capability.Configuration.SetCommunicationPort {
+		baseReconciler.AddOnAfterStatefulSetCreateListener(setCommunicationsPort(instance))
+	}
+
+	if capability.Configuration.SetReadinessPort {
+		baseReconciler.AddOnAfterStatefulSetCreateListener(setReadinessProbePort(instance))
```

Comment:
@meik99 Can you explain why it is called _OnAfter..._? IMO _OnStatefulSetCreateListener_ already explains what it is.

Reply:
Because the listener is executed after the stateful set is created, meaning that there are no more significant changes made to it (other than calculating the hash). I wanted to make that visible, since `OnStatefulSetCreateListener` does not specify when exactly it is called. The naming also allows adding a `OnBefore...` listener if needed